RationaleMCP/0031/differences.md

```diff
@@ -81,3 +81,22 @@ This covers what one can currently express in full Modelica.  In the future, one
 
 ## Array dimensions with parameter variability
 In Flat Modelica, an array dimension is allowed to have `parameter` variability, that is, the dimension isn't known until after solving the initialization problem in the simulation.  _TO BE ELABORATED…_
+
+## Subscripting of general expressions
+In Flat Modelica it is possible to have a subscript on any (parenthesized) expression.
+The reason for this generalization is that some manipulations, in particular inlining of function calls, can lead to such 
+expressions and without the slight generalization we could not generate flat Modelica for them. It does not add any real complication
+to the translator.
+
+The reason it is restricted is that `a.x[1]` (according to normal Modelica semantics) and `(a.x)[1]` will often work differently.
+Consider
+```
+record R
+  Real x[2];
+end R;
+R a[3];
+```
+Here  `a.x[1]` will cause slicing generating the array  `{a[1].x[1],a[2].x[1],a[3].x[1]}`, whereas `(a.x)[1]` generates the array 
```

Comment:
I'd consider both variants to be slices, so I'd say something like:
> Here, `a.x[1]` is a slice of the `x`, resulting in …, whereas `(a.x)[1]` is a slice of `a.x`, resulting in …

It could also be worth noting that giving array subscripting higher precedence than member access doesn't solve the problem, as `x[1]` cannot be evaluated as an expression; the `x` only makes sense as a member of `a`.

Reply:
It's correct that both can be viewed as slices, and I now tried to follow the Modelica spec calls `a.x` and `a.x[1]` slice operations (10.6.9).